### ISSUE TITLE
Patching of configs.imageregistry.operator.openshift.io times out

### ISSUE THREAD

**Issue by EmmanuelKasper** [RedHat-EMEA-SSA-Team/hetzner-ocp4#173]
Hi
With `nfs-storage: true` in `cluster.yml` the ansible playbook `post-install-storage-nfs.yml` times out at the patch command:

~~~
- name: Add pvc registry-storage to image registry
  command: |
    oc patch configs.imageregistry.operator.openshift.io cluster --type='json' -p='[{"op": "remove", "path": "/spec/storage" },{"op": "add", "path": "/spec/storage", "value": {"pvc":{"claim": "registry-storage"}}}]' --kubeconfig {{ openshift_install_dir }}/auth/kubeconfig
  register: registry_status
  retries: 60
  delay: 30
  until: registry_status.rc == 0
  tags:
   - storage
~~~

after the timeout, the status of the image-registry operator is as such:
oc describe co image-registry
~~~
  conditions:
  - lastTransitionTime: "2021-04-02T08:43:41Z"
    message: |-
      Available: Error: storage backend not configured
      ImagePrunerAvailable: Pruner CronJob has been created
    reason: StorageNotConfigured
    status: "False"
    type: Available
  - lastTransitionTime: "2021-04-02T08:34:19Z"
    message: 'Progressing: Unable to apply resources: storage backend not configured'
    reason: Error
    status: "False"
    type: Progressing
  - lastTransitionTime: "2021-04-02T08:43:41Z"
    message: 'Degraded: Error: storage backend not configured'
    reason: StorageNotConfigured
    status: "True"
    type: Degraded
~~~

I was able to manually fix this by simply adding the pvc manually with `oc edit configs.imageregistry.operator.openshift.io/cluster`

~~~
    pvc:
      claim: registry-storage

~~~

could it be that having a `remove` and a `add` action in the oc path command confuses the API server ?

on a side node, how I can reapply the hetzner-ocp4/ansible/roles/openshift-4-cluster/tasks/post-install-storage-nfs.yml
playbook without reboostrating the whole cluster each time ?

**Comment by EmmanuelKasper** [comment]
OK I found out the reason `oc` was failing is imply ansible could not find the binary (this explains too why I always had to manually sign the CSR myself )
The patch at https://gist.github.com/EmmanuelKasper/0563efab29127fc187a784af4ee6e659
should fix the issue, to be applief with `git am`.

**Comment by rbo** [comment]
Please update the oc command 'xargs --no-run-if-empty oc...' too.

Check out the ansible tags there some tags to run only post install. Or you simple run 02-create-cluster again, that works very well.

On Tuesday I will apply your patch and can provide some ansible Tag examples.

Thanks for contributions!

**Comment by EmmanuelKasper** [comment]
Thanks 02-create-cluster worked well. +1 for idempotents manifests !
I actualized the patch at https://gist.github.com/EmmanuelKasper/0563efab29127fc187a784af4ee6e659
so that it covers the xargs loop too.

**Comment by rbo** [comment]
Everything is applied into branch `absolute-to-oc` I fixed the xargs this morning. Test installation is running.

**Comment by rbo** [comment]
Merged into devel will be merged into master with #172
